services/web-app/utils/sendLocalRequest.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { getRunMode, RunMode } from '@carbon-platform/api/runtime'
+import cookie from 'cookie'
+
+/**
+ * Retrieves and returns serialized auth cookie
+ *
+ * @param {NextRequest} req getServerSideProps request object
+ * @returns {string} serialized auth cookie
+ */
+function getAuthCookie(req) {
+  const sessionCookie = req.cookies['connect.sid']
+  return sessionCookie ? cookie.serialize('connect.sid', sessionCookie) : ''
+}
+
+/**
+ * Sends an http request internally
+ * @param {NextRequest} req - Original request object that came through the middleware
+ * or getServerSideProps.context
+ * @param {string} path - Relative path to request
+ * @param {boolean} forwardAuthCookie - Should auth cookie be forwarded to local request
+ * @param {string} method - Request method to use, default to 'GET'
+ * @param {Object} body - Body object to send to fetch command
+ * @param {Object} requestOptions - Additional request options to send to the fetch command
+ * @returns {Promise<Response>} Promise return from fetch()
+ */
+export default function sendLocalRequest(
+  req,
+  path,
+  forwardAuthCookie = false,
+  method = 'GET',
+  body = {},
+  requestOptions = {}
+) {
+  const protocol =
+    getRunMode() === RunMode.Prod || process.env.RUNNING_SECURELY === '1' ? 'https' : 'http'
+
+  const requestUrl = `${protocol}://localhost:${req.nextUrl.port}/${path}`
+
+  const reqOptions = {
+    method,
+    headers: {
+      'Content-Type': 'application/json',
+      cookie: forwardAuthCookie ? getAuthCookie(req) : ''
+    },
+    ...requestOptions
+  }
+
+  if (method !== 'GET') {
+    reqOptions.body = JSON.stringify(body)
+  }
```

Comment:
It might be easier to default `body` to null. If it's found, then set it in the reqOptions, if not, don't set the variable. There's a bunch of http request types that might need a body, so rather than enumerating them all here, just delegate that to the caller.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods

(If you click on each one, it'll tell you whether it needs a request body)

Reply:
Done!